src/main/java/command/LyricCommand.java

```diff
@@ -0,0 +1,113 @@
+import service.music.GuildMusicManager;
+import service.music.HelpProcess;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+import java.util.concurrent.CompletableFuture;
+
+public class LyricCommand extends Command
+{
+    private final NanoClient client;
+
+    public LyricCommand(NanoClient client)
+    {
+        this.client = client;
+
+        this.name = "lyric";
+        this.arguments = "<query>";
+        this.help = "Get lyric for music.\n" +
+                    "If `query` is empty, the title of the currently playing music will be used.\n";
+        this.guildOnly = true;
+        this.category = new Category("Music");
+        this.help = HelpProcess.getHelp(this);
+    }
+
+    @Override
+    protected void execute(CommandEvent event)
+    {
+        EmbedBuilder embed = new EmbedBuilder();
+        embed.setColor(event.getMember().getColor());
+        GuildMusicManager musicManager = this.client.getGuildAudioPlayer(event.getGuild());
+        String query = "";
+
+        if (event.getArgs().trim().equals("") && musicManager.player.getPlayingTrack() == null)
+        {
+            embed.setTitle("Failed");
+            embed.addField(
+                    ":x:",
+                    "Can't search lyric, because `query` is empty and currently no playing music.",
+                    true);
+            event.reply(embed.build());
+            return;
+        }
+        else if (event.getArgs().trim().equals("") && musicManager.player.getPlayingTrack() != null)
+        {
+            query = musicManager.player.getPlayingTrack().getInfo().title;
+        }
+        else
+        {
+            query = event.getArgs();
+        }
+
+        String finalQuery = query;
+
+        CompletableFuture.runAsync(() ->
+        {
+            try
+            {
+                URL url = new URL("https://lyrics.tsu.sh/v1/?q=" + URLEncoder.encode(finalQuery, StandardCharsets.UTF_8.toString()));
+                HttpURLConnection http = (HttpURLConnection) url.openConnection();
+                http.setRequestProperty("User-Agent", "Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.4; en-US; rv:1.9.2.2) Gecko/20100316 Firefox/3.6.2");
+                http.setRequestProperty("accept", "application/json");
+
+                StringBuilder output = new StringBuilder();
+                try (Reader reader = new BufferedReader(new InputStreamReader(http.getInputStream(), Charset.forName(StandardCharsets.UTF_8.name()))))
+                {
+                    int c = 0;
+                    while ((c = reader.read()) != -1)
+                    {
+                        output.append((char) c);
+                    }
+                }
```

Comment:
using `YoutubeSearchApi.Utils.httpRequest()` will throw http error 403

Reply:
![image](https://user-images.githubusercontent.com/29156554/98314624-fa885d00-2008-11eb-89f2-b7a0474a4858.png)

here the proof.

Reply:
Update the method's parameter, so it can accept the request property. or just use a bool, whether use request property or not

https://github.com/Shiroechi/YouTubeSearchApi-Java/blob/82d47dfd9a173647c20ec89e9ff663159acf9a58/src/main/java/YouTubeSearchApi/utility/Utils.java#L21-L43

Reply:
done

Reply:
Have you tested the `search` method after the `httpRequest()` method updates ?

Reply:
@Shiroechi

Reply:
nooo

Reply:
user agent property is to let the server know from what client the user access.
So it will still work, if not work than it is not user agent fault i think

Reply:
dude, remember dis issue ? 

https://github.com/Shiroechi/YouTubeSearchApi-Java/commit/82d47dfd9a173647c20ec89e9ff663159acf9a58

just test it

Reply:
> user agent property is to let the server know from what client the user access.
> So it will still work, if not work than it is not user agent fault I think.
no. if the server's backend does not support the **user agent** sent by the client browser, the server will response with error.

Reply:
fuckk, youtube errorrrrrrr

Reply:
> edit:
> and i think it is not good to use static method `httpRequest`. what i know its not a thread safe.

No. `httpRequest()` method is stateless (it only uses local variables and the argument), therefore is thread safe 

local variables, method parameters, and return values are inherently "thread safe."

But the instance variable in method parameter might not be thread safe (different discussion, depends on the instance, is it mutable or not, if its mutable then its not thread safe)

SO, `All methods are thread safe. UNTIL they change the state of some static variable or instance member variable` and this is not the case for `httpRequest` method. 
- It does not use any instances as argument or change any static variable. 
- It only uses local variables and the argument (the argument itself is from local variables).

> https://codeflex.co/java-multithreading-completablefuture-explained/

This article does not explain any about thread safety. 

> 
> edit:
> yeshhh, using latest chrome user agent, and there's no error

Does it work on all the test case ?

Reply:
yes it works, when i test it